tests/PHPStan/Analyser/nsrt/preg_match_shapes.php

```diff
@@ -768,3 +771,125 @@ function bug11604b (string $string): void {
 		assertType("array{0: string, 1?: ''|'XX', 2?: ''|'YY', 3?: 'ZZ'}", $matches);
 	}
 }
+
+function testLtrimDelimiter (string $string): void {
+	if (preg_match(' /(x)/', $string, $matches)) {
+		assertType("array{string, 'x'}", $matches);
+	}
+
+	if (preg_match('  /(x)/', $string, $matches)) {
+		assertType("array{string, 'x'}", $matches);
+	}
+}
+
+function testUnescapeBackslash (string $string): void {
+	if (preg_match(<<<'EOD'
+		~(\[)~
+		EOD, $string, $matches)) {
+		assertType("array{string, '['}", $matches);
+	}
+
+	if (preg_match(<<<'EOD'
+		~(\d)~
+		EOD, $string, $matches)) {
+		assertType("array{string, numeric-string}", $matches);
+	}
+
+	if (preg_match(<<<'EOD'
+		~(\\d)~
+		EOD, $string, $matches)) {
+		assertType("array{string, '\\\d'}", $matches);
+	}
+
+	if (preg_match(<<<'EOD'
+		~(\\\d)~
+		EOD, $string, $matches)) {
+		assertType("array{string, non-falsy-string}", $matches);
+	}
+
+	if (preg_match(<<<'EOD'
+		~(\\\\d)~
+		EOD, $string, $matches)) {
+		assertType("array{string, '\\\\\\\d'}", $matches);
+	}
+}
+
+function testEscapedDelimiter (string $string): void {
+	if (preg_match(<<<'EOD'
+		/(\/)/
+		EOD, $string, $matches)) {
+		assertType("array{string, '/'}", $matches);
+	}
+
+	if (preg_match(<<<'EOD'
+		~(\~)~
+		EOD, $string, $matches)) {
+		assertType("array{string, '~'}", $matches);
+	}
+
+	if (preg_match(<<<'EOD'
+		~(\[2])~
+		EOD, $string, $matches)) {
+		assertType("array{string, '[2]'}", $matches);
+	}
+
+	if (preg_match(<<<'EOD'
+		[(\[2\])]
+		EOD, $string, $matches)) {
+		assertType("array{string, '[2]'}", $matches);
+	}
+
+	if (preg_match(<<<'EOD'
+		~(\{2})~
+		EOD, $string, $matches)) {
+		assertType("array{string, '{2}'}", $matches);
+	}
+
+	if (preg_match(<<<'EOD'
+		{(\{2\})}
+		EOD, $string, $matches)) {
+		assertType("array{string, '{2}'}", $matches);
+	}
+
+	if (preg_match(<<<'EOD'
+		~([a\]])~
+		EOD, $string, $matches)) {
+		assertType("array{string, 'a'}", $matches); // should be array{string, ']'|'a'}
+	}
+
+	if (preg_match(<<<'EOD'
+		~([a[])~
+		EOD, $string, $matches)) {
+		assertType("array{string, '['|'a'}", $matches);
+	}
+
+	if (preg_match(<<<'EOD'
+		~([a\]b])~
+		EOD, $string, $matches)) {
+		assertType("array{string, 'a'|'b'}", $matches); // should be array{string, ']'|'a'|'b'}
```

Comment:
I feel like these should be fixed as it's clearly a bug and not just a not-as-specific-as-it-could-be result.

Reply:
done - https://github.com/phpstan/phpstan-src/pull/3488/commits/6eed070cef2f04d7ca90557dd85f4fbb93f97968